src/fast_check/transform_dts.rs

```diff
@@ -0,0 +1,730 @@
+    }
+  }
+
+  fn class_body_to_type(&mut self, body: &[ClassMember]) -> Vec<ClassMember> {
+    body
+      .iter()
+      .filter_map(|member| match member {
+        ClassMember::Constructor(class_constructor) => {
+          let mut new_constructor = class_constructor.clone();
+          new_constructor.body = None;
+          Some(ClassMember::Constructor(new_constructor))
+        }
+        ClassMember::Method(method) => {
+          let mut new_method = method.clone();
+          new_method.function.body = None;
+          if new_method.kind == MethodKind::Setter {
+            new_method.function.return_type = None;
+          }
+          Some(ClassMember::Method(new_method))
+        }
+        ClassMember::ClassProp(prop) => {
+          let mut new_prop = prop.clone();
+          if new_prop.type_ann.is_none() {
+            if let Some(value) = new_prop.value {
+              new_prop.type_ann = self
+                .expr_to_ts_type(*value, false)
+                .map(type_ann)
+                .or(Some(any_type_ann()));
+            }
+          }
+          new_prop.value = None;
+
+          Some(ClassMember::ClassProp(new_prop))
+        }
+        ClassMember::TsIndexSignature(index_sig) => {
+          Some(ClassMember::TsIndexSignature(index_sig.clone()))
+        }
+
+        _ => None,
+      })
+      .collect()
+  }
+}
+
+#[cfg(test)]
+mod tests {
+  use crate::{
+    fast_check::{
+      transform::{emit, CommentsMut},
+      transform_dts::FastCheckDtsTransformer,
+    },
+    source::{MemoryLoader, Source},
+    symbols::RootSymbol,
+    BuildOptions, DefaultModuleParser, GraphKind, ModuleGraph,
+  };
+  use url::Url;
+
+  async fn transform_dts_test(source: &str, expected: &str) {
+    let specifier = Url::parse("file:///mod.ts").unwrap();
+
+    let spec_str = specifier.to_string();
+    let mut loader = MemoryLoader::new(
+      vec![(
+        spec_str.clone(),
+        Source::Module {
+          specifier: spec_str,
+          maybe_headers: None,
+          content: source.to_string(),
+        },
+      )],
+      vec![],
+    );
+    let parser = DefaultModuleParser {};
+    let mut graph = ModuleGraph::new(GraphKind::All);
+    graph
+      .build(
+        vec![specifier.clone()],
+        &mut loader,
+        BuildOptions {
+          module_parser: Some(&parser),
+          ..Default::default()
+        },
+      )
+      .await;
+
+    let root_sym = RootSymbol::new(&graph, &parser);
+
+    let module_info = root_sym
+      .module_from_specifier(&specifier)
+      .unwrap()
+      .esm()
+      .unwrap();
+
+    let module = module_info.source().module().clone();
+
+    let mut transformer = FastCheckDtsTransformer::new();
+    let module = transformer.transform(module).unwrap();
+
+    let parsed_source = module_info.source();
+    let comments =
+      CommentsMut::new(parsed_source.comments().as_single_threaded());
+    let comments2 = comments.into_multi_threaded().into_single_threaded();
```

Comment:
I think we can just do `let comments = parsed_source.comments().as_single_threaded()`?

Reply:
I'm afraid not, it leads to a type mismatch:

```sh
mismatched types
expected reference `&SingleThreadedComments`
   found reference `&CommentsMut`
```

Reply:
What does the full code look like? There shouldn't be a `CommentsMut` in there.

Reply:
ohh wait you're right, I can drop the whole `CommentsMut` thing. Thanks for the tip 👍